src/Microsoft.ML.OnnxTransformer/OnnxSessionOptions.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.ML.Data;
+using Microsoft.ML.OnnxRuntime;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Transforms.Onnx;
+
+namespace Microsoft.ML.Transforms.Onnx
+{
+    public static class OnnxSessionOptionsExtensions
+    {
+        private const string OnnxSessionOptionsName = "OnnxSessionOptions";
+
+        public static OnnxSessionOptions GetOnnxSessionOption(this IHostEnvironment env)
+        {
+            if (env is IHostEnvironmentInternal localEnvironment)
+            {
+                return localEnvironment.GetOptionOrDefault<OnnxSessionOptions>(OnnxSessionOptionsName);
+            }
+
+            throw new ArgumentException("No Onnx Session Options");
+        }
+
+        public static void SetOnnxSessionOption(this IHostEnvironment env, OnnxSessionOptions onnxSessionOptions)
+        {
+            if (env is IHostEnvironmentInternal localEnvironment)
+            {
+                localEnvironment.AddOrOverwriteOption(OnnxSessionOptionsName, onnxSessionOptions);
+            }
+
+            throw new ArgumentException("No Onnx Session Options");
+        }
+    }
+
+    public class OnnxSessionOptions
```

Comment:
>OnnxSessionOptions [](http://example.com/codeflow?start=17&length=18)

- make it sealed
- Is it intended to have public fields instead of properties here?

Reply:
Its now sealed.

So our other Options classes just use public fields instead of properties which is why I went that route. Do you know why that choice may have been made initially?

Reply:
I have no idea why fields are used in the first place, but this is against the design guidelines. I would suggest not using fields and moving forward we apply the guidelines. 

https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/field